app/models/dataset/DatasetService.scala

```diff
@@ -69,6 +71,18 @@ class DatasetService @Inject()(organizationDAO: OrganizationDAO,
     createDataset(dataStore, organizationName, unreportedDatasource)
   }
 
+  def getAllNotYetUploadedDatasetOfUser(userId: ObjectId, organizationId: ObjectId)(
+      implicit ctx: DBAccessContext): Fox[List[DatasetCompactInfo]] =
+    datasetDAO.findAllCompactWithSearch(
+      uploaderIdOpt = Some(userId),
+      organizationIdOpt = Some(organizationId),
+      isActiveOpt = Some(false),
+      includeSubfolders = true,
+      statusOpt = Some(notYetUploadedStatus),
+      // Only list pending upload since the two last weeks.
+      createdSinceOpt = Some(Instant.now - Duration(14, TimeUnit.DAYS))
```

Comment:
If you import scala.concurrent.duration._ you can write
```suggestion
      createdSinceOpt = Some(Instant.now - 14 days))
```